Adds user-facing quickstart for Ensign (#100)
rotationalio/ensign: docs/content/eventing/_index.md

Brokers are also in charge of things like keeping events in the correct order, remembering which subscribers are listening to a topic stream, recording the last message each subscriber retrieved, etc.

In Ensign, brokers can save events permanently even after they have been retrieved (to support "time travel" &mdash; the ability to retroactively scan through an event stream to support analytics and machine learning).

#### **client** <a name="client"></a>
In order to write or read data from an underlying data system (like a database or event stream), you need a client to connect to the data system and interact with it as needed (such as reading and writing data). This connection often looks something like `conn = DBConnection(credentials)`, and after creating the `conn` variable, subsequent lines of code can leverage it to perform the kinds of data interactions you wish to make.

In Ensign, to establish a client, you need an [API key](#api-key).

```golang
import 	(
    ensign "github.com/rotationalio/ensign/sdks/go"
)

> client, _ := ensign.New(&ensign.Options{

Reviewer: I wonder if we should discourage "throwing the error to the underscore" and define an `err` here as well. I don't think we have to have error handling code here because it's nice to advertise the simplicity but I do think it's good to at least hint what the other thing being returned is.

Additionally if someone is copying the code from here and copies in the wrong key or something, an error is way more helpful than a nil panic.

In a microservices context, we can reduce application latency by using asynchronous communications and parallelizing functions so they run faster. Network latency can be reduced by creating more efficient communications between servers (e.g. using more scalable consensus algorithms).

#### **microservice** <a name="microservice"></a>
A microservice is a computer application composed of a collection of lightweight services, each of which is responsible for some discrete task.

Microservices can be coordinated to communicate via [events](#events).

#### **mime type**
A MIME (Multipurpose Internet Mail Extensions) type is a label that identifies a type of data, such as CSV, HTML, JSON, or protocol buffer.

MIME types allow an application to understand how to handle incoming and outgoing data.

#### **organization**
An Ensign organization is a collection of users who are working under the same Ensign [tenant](#tenant).

#### **publisher** <a name="publisher"></a>
In an event-driven microservice, a publisher is responsible for emitting [events](#events) to a [topic stream](#topic).

In Ensign, you can create a publisher once you have established a [client](#client):

```golang

> pub, _ := client.Publish(...)

Reviewer: Similarly should we specify what the other field being returned is or leave it to the API docs?

```

#### **real-time**
This is a tricky one because real-time can be used to mean different things. In some cases, "real-time" is used as a synonym for synchronous (i.e. the opposite of [asynchronous](#asynchronous)). However, the term is also used to mean "very fast" or ["low latency"](#latency).

#### **sdk**
SDK stands for "Software Development Kit". Software applications designed for a technical/developer audience frequently are considerate enough to provide user-facing SDKs in a few languages (e.g. Golang, Python, JavaScript). These SDKs give users a convenient way to interact with the application using a programming language with which they are familiar.

> Ensign currently offers two SDKs: the Golang SDK and a Watermill API-compatible SDK.

Reviewer: Should we link to the SDKs here? Either the docs pages or the repos?

#### **stream**
An event stream is a flow composed of many, many individual pieces of data called [events](#events).

#### **subscriber**

> In an event-driven context, a subscriber is a downstream component that is listening for [events](#events) published by a [publisher](#publisher) on to a [topic](#topic).

Reviewer: ```suggestion
In an event-driven context, a subscriber is a downstream component that is listening for [events](#events) published by a [publisher](#publisher) onto a [topic](#topic).
```
